packages/serverpod/lib/src/server/future_call_manager.dart

```diff
@@ -147,12 +158,15 @@ class FutureCallManager {
 
     // If we are running as a maintenance task, we shouldn't check the queue
     // again.
-    if (_server.serverpod.commandLineArgs.role == ServerpodRole.monolith) {
+    if (_server.serverpod.commandLineArgs.role == ServerpodRole.monolith &&
+        !_shuttingDown) {
       // Check the queue again in 5 seconds
       _timer = Timer(const Duration(seconds: 5), _checkQueue);
     } else if (_server.serverpod.commandLineArgs.role ==
         ServerpodRole.maintenance) {
       onCompleted();
     }
+
+    pendingFutureCall.complete();
```

Comment:
Question: Any reason for not nulling this variable after the complete?

Reply:
The variable is non nullable now so it does not apply.